common/models/role.js

```diff
@@ -29,44 +33,66 @@ module.exports = function(Role) {
-          if (callback) callback(err);
-          return;
-        }
-        return mappings.map(function(m) {
-          return m.principalId;
-        });
-      });
-    };
+    var principalTypesToModels = {};
+
+    principalTypesToModels[RoleMapping.USER] = User;
+    principalTypesToModels[RoleMapping.APPLICATION] = Application;
+    principalTypesToModels[RoleMapping.ROLE] = Role;
+
+    Object.keys(principalTypesToModels).forEach(function(principalType){
+      var model = principalTypesToModels[principalType];
+      var pluralName = model.pluralModelName.toLowerCase();
+      /** 
+       * Fetch all users assigned to this role
+       * @function Role.prototype#users
+       * @param {object} [query] query object passed to model find call
+       * @param  {Function} [callback]
+       */ 
+      /** 
+       * Fetch all applications assigned to this role
+       * @function Role.prototype#applications
+       * @param {object} [query] query object passed to model find call
+       * @param  {Function} [callback]
+       */
+      /** 
+       * Fetch all roles assigned to this role
+       * @function Role.prototype#roles
+       * @param {object} [query] query object passed to model find call
+       * @param {Function} [callback]
+       */ 
+      Role.prototype[pluralName] = function(query, callback) {
+        listByPrincipalType(model, principalType, query, callback);
+      }
+    });
 
-    Role.prototype.applications = function(callback) {
-      roleMappingModel.find({where: {roleId: this.id,
-        principalType: RoleMapping.APPLICATION}}, function(err, mappings) {
-        if (err) {
-          if (callback) callback(err);
-          return;
-        }
-        return mappings.map(function(m) {
-          return m.principalId;
-        });
-      });
-    };
+    /**
+     * Fetch all models assigned to this role
+     * @param {*} model model type to fetch
+     * @param {String} [principalType] principalType used in the rolemapping for model
+     * @param {object} [query] query object passed to model find call
+     * @param  {Function} [callback]
+     */
+    function listByPrincipalType(model, principalType, query, callback) {
+      if (callback === undefined) {
+        callback = query;
+        query = {};
+      }
 
-    Role.prototype.roles = function(callback) {
-      roleMappingModel.find({where: {roleId: this.id,
-        principalType: RoleMapping.ROLE}}, function(err, mappings) {
+      roleMappingModel.find({
+        where: {roleId: this.id, principalType: principalType}
+      }, function(err, mappings) {
+        var ids;
         if (err) {
-          if (callback) callback(err);
-          return;
+          return callback(err);
         }
-        return mappings.map(function(m) {
+        ids = mappings.map(function(m) {
           return m.principalId;
         });
+        query.where = query.where || {};
+        query.where.id = {inq: ids};
+        model.find(query, callback);
```

Comment:
Please be specific about the arguments you are passing to `callback`. Your current implementation creates a contract saying `listByPrincipalType` passes the same callback arguments as `PersistedModel.find`, which may become difficult to preserve in future refactorings.

Also the jsdoc entry should include description of callback arguments.